document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresQueryBuilder.java

```diff
@@ -0,0 +1,39 @@
+package org.hypertrace.core.documentstore.postgres;
+
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.DOCUMENT;
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.ID;
+
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.hypertrace.core.documentstore.model.subdoc.SubDocumentValue;
+import org.hypertrace.core.documentstore.postgres.subdoc.PostgresSubDocumentValueParser;
+
+@RequiredArgsConstructor
+public class PostgresQueryBuilder {
+  private final String collectionName;
+  private final PostgresSubDocumentValueParser subDocValueParser =
+      new PostgresSubDocumentValueParser();
+
+  public String buildSelectQueryForUpdateSkippingLocked(
+      org.hypertrace.core.documentstore.postgres.query.v1.PostgresQueryParser parser) {
+    final String selections = parser.getSelectionsWithImplicitId();
```

Comment:
We are not expecting a query with `unnest expression` for update API right?
If, required, let's add it when needed.

Reply:
That's right.